### ISSUE TITLE
Pyright not getting installed if setup overridden by lspconfig customizations

### ISSUE THREAD

**Issue by smacz42** [williamboman/nvim-lsp-installer#304]
Similar to[An iissue Originally posted by @HagaiHargil](https://github.com/williamboman/nvim-lsp-installer/issues/276#issuecomment-977734451), after installing Pyright I get the issue:

```
Error detected while processing FileType Autocommands for "python":                                                                                                                                                                           
E5108: Error executing lua /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:384: start `pyright-langserver` failed: ENOENT: no such file or directory                                                                                              
```

I have this in my config where I have lspconfig installed:

```
local lsp = require('lspconfig')

lsp.pyright.setup{
    settings = {
        python = {
            analysis = {
                extraPath = {'.'},
                autoImportCompletions = true,
            }
        }
    }
}
```

This is how I _believe_ it is suggested to override configs. However, when I comment this out I do NOT get the error.

I believe that the nvim-lsp-installer sets the path by setting the `cmd` attribute using variables in this plugin's scope, that may not be available in the scope for lspconfig.

A quick fix here is to add right in the top level, the full path to the command to run:

```
cmd = {os.getenv("HOME") .. '/.local/share/nvim/lsp_servers/python/node_modules/.bin/pyright-langserver', '--stdio'},
```

However, I know that this plugin tries to set that up. Seems a waste to have to override it if there is any customiation to the config.

**Comment by williamboman** [comment]
Hello! I just wrote an answer to a similar issue here https://github.com/williamboman/nvim-lsp-installer/issues/305#issuecomment-986030260, hopefully it makes sense. tl;dr is that the `cmd` is only automatically injected if you set up the server via nvim-lsp-installer.

**Comment by williamboman** [comment]
And here's an example snippet in the wiki https://github.com/williamboman/nvim-lsp-installer/wiki/Advanced-Configuration#overriding-the-default-lsp-server-options

**Comment by smacz42** [comment]
Ok, that works.

Just so you know where I was coming from (and in case anyone else gets tripped up by this), I had installed [Plug Config](https://github.com/ouuan/vim-plug-config), whose paradigm is to put all associated configs settings in files separated by plugin.

So, when I went to override the lspconfig, I went to put it in a separate file from the installer setup. However, since these two plugins are so closely intertwined, it makes sense in this situation to break that paradigm and overlap those plugin configs.
